tensorflow/models/rnn/ptb/ptb_word_lm.py

```diff
@@ -239,26 +239,39 @@ class TestConfig(object):
   vocab_size = 10000
 
 
-def run_epoch(session, m, data, eval_op, verbose=False):
+def run_epoch(session, model, data, eval_op, verbose=False):
   """Runs the model on the given data."""
-  epoch_size = ((len(data) // m.batch_size) - 1) // m.num_steps
+  epoch_size = ((len(data) // model.batch_size) - 1) // model.num_steps
   start_time = time.time()
   costs = 0.0
   iters = 0
-  state = m.initial_state.eval()
-  for step, (x, y) in enumerate(reader.ptb_iterator(data, m.batch_size,
-                                                    m.num_steps)):
-    cost, state, _ = session.run([m.cost, m.final_state, eval_op],
-                                 {m.input_data: x,
-                                  m.targets: y,
-                                  m.initial_state: state})
+  state = []
+  for c, m in model.initial_state: # initial_state: ((c1, m1), (c2, m2))
+    state.append((c.eval(), m.eval()))
+  for step, (x, y) in enumerate(reader.ptb_iterator(data, model.batch_size,
+                                                    model.num_steps)):
+    fetches = []
+    fetches.append(model.cost)
```

Comment:
i think you can do:

fetches = [model.cost, model.final_state, eval_op]

then

cost, state, _ = session.run(fetches, feed_dict)

though i wouldn't use state since it conflicts with the variable state above.